doc/design/draft/subscription.md

```diff
@@ -0,0 +1,180 @@
+
+The subscriber cache contains the one sample ordered after _DeliveryTime_.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+
+#### Subscription with _HistorySize_ 6
+
+The subscriber cache contains the six samples ordered after _DeliveryTime_.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+| P2        | 24        | H      |
+| P2        | 22        | G      |
+
+#### Subscription with _HistorySize_ 20
+
+The subscriber cache contains the ten samples ordered after _DeliveryTime_.
+It contains only ten since there are no more samples available on publisher side.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+| P2        | 24        | H      |
+| P2        | 22        | G      |
+| P1        | 21        | C      |
+| P2        | 20        | F      |
+| P1        | 11        | B      |
+| P1        | 1         | A      |
+
+### Subscribing with concurrent send
+
+Let's assume publisher P1 would like to deliver the sample K at timepoint
+52 while the subscriber requested subsription. Furthermore, the subscriber
+requests always a _HistorySize_ of 2.
+
+All the listed cases should behave identical if there is more then one
+concurrent publish in progress.
+
+#### Publisher before _DeliveryTime_
+
+The subscriber cache contains two samples.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+
+The sample K will be delivered after successful subscription.
+
+#### Publisher after _DeliveryTime_
+
+The subscriber cache contains two samples.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P1        | 51        | K      |
+| P2        | 42        | J      |
+
+The sample K will not be delivered by the publisher since the _DeliveryTime_
+is defined as exactly the timepoint after the sample is placed into the ChunkHistory
+and the subscriber cache is based on that ChunkHistory.
+
+### Subscribing After Publisher Stops Offering
+
+In this case we assume that publisher P1 has stopped offering the service
+before the subscription was requested.
+
+#### Subscription with _HistorySize_ 4
+
+The subscriber cache contains the four samples ordered after _DeliveryTime_.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+
+
+
+## Design
+
+### Idea 1: One Ring Buffer Which Contains The Samples
+
+- All publishers are sharing one random access lock-free ring-buffer.
```

Comment:
This would essentially require a muli write, multi read lock-free buffer. Possible, but more difficult then the single-write multi read buffer I propose in https://github.com/eclipse-iceoryx/iceoryx/issues/857. This is also quite a bit further away from our current archtitecture since we need such a buffer per service/topic and not per publisher then. 

I propose we start in small steps and properly implement the buffering in our current approach - buffering on publisher side. Then we can move to the more complicated solution where we buffer across publishers in a centralized manner per topic.

Reply:
@MatthiasKillat  you are right, but here we just wanted to collect ideas how the final solution could look like since we need something where we can converge to when we start with small steps.